### PR TITLE
Adds libghthash formula

### PR DESCRIPTION
**Pull request by oyvindrobertsen**
Libghthash is a lightweight, portable and extendable hash table implementation.

### CODE REVIEW

File: Library/Formula/libghthash.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+        int main() {
+            ght_hash_table_t* table = ght_create(1);
+            ght_finalize(table);
+            return 0;
+        }
+      EOS
+      system ENV.cc "test.c", "-I#{include}", "-L#{lib}", "-lghthash", "-o", "test"
+      system "./test"
+    end
```
Comment (tdsmith) [inline Library/Formula/libghthash.rb]: Great test; thanks.

File: Library/Formula/libghthash.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Libghthash < Formula
+  homepage "http://www.bth.se/people/ska/sim_home/libghthash.html"
+  url "http://www.bth.se/people/ska/sim_home/filer/libghthash-0.6.2.tar.gz"
+  version "0.6.2"
```
Comment (tdsmith) [inline Library/Formula/libghthash.rb]: `brew audit` will probably complain that the version is correctly parsed from the URL so the `version` line is redundant and should be removed.

**Comment by oyvindrobertsen** [comment]
Thanks! I've removed the version line in https://github.com/oyvindrobertsen/homebrew/commit/d29be556413050eb9b550b2429498431881a7fbe .
`brew audit` also complained that non-libraries were installed to lib, specifically the autogenerated documentation for the library. I don't have control over the source of the library, so there's not much I can do about that.

**Comment by DomT4** [comment]
Does anything break if you ` mv ` the documentation out of ` lib ` and into ` share/"doc" ` or similar?

File: Library/Formula/libghthash.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+      (testpath/"test.c").write <<-EOS.undent
```
Comment (mikemcquaid) [inline Library/Formula/libghthash.rb]: Use a two-character indent in this function, thanks!

**Comment by oyvindrobertsen** [comment]
@mikemcquaid I assumed you were only talking about the ruby parts of the test :)
@DomT4 Haven't tried that, any suggestions as to where/how in the formula that should be done?

**Comment by DomT4** [comment]
@oyvindrobertsen Something like:

```ruby
    system "make", "install"
    mv Dir["lib/*.txt"], share/"doc"
  end
```

Or:

```ruby
    system "make", "install"
    doc.install Dir["lib/*.txt"]
  end
```

Should work. Replace the ` *.txt ` with whatever file extension the documentation is. If it's man documentation it'd be something like ` man1.install Dir["lib/*.1"] ` and so on. If there's no file extension to the documentation might have to try a glob or something similar, but if simple works that'd be nice :smile_cat:

**Comment by oyvindrobertsen** [comment]
Thanks @DomT4! `brew audit` has stopped complaining now, crossing fingers :)

File: Library/Formula/libghthash.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  url "http://www.bth.se/people/ska/sim_home/filer/libghthash-0.6.2.tar.gz"
+  sha1 "436146998c30a31b83adf4f58d978fccd78f2ba5"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+    doc.install Dir["#{lib}/*[^.a]"]
```
Comment (tdsmith) [inline Library/Formula/libghthash.rb]: libghthash doesn't build shared libraries?

Comment (oyvindrobertsen) [inline Library/Formula/libghthash.rb]: Not that I'm aware of, no.

**Comment by tdsmith** [comment]
I don't see that it installs any autogenerated docs to lib? I make installed to /tmp/libghthash:

```
/tmp/libghthash/lib/libghthash
/tmp/libghthash/lib/libghthash.1
/tmp/libghthash/lib/libghthash.1.1.0
```

These are all dylibs; they're missing the .dylib extension. The formula should fix the build script to write the names correctly or rename them (to i.e. /tmp/libghthash/lib/libghthash.1.1.0.dylib) after building them.

Are you seeing something else?

**Comment by oyvindrobertsen** [comment]
Ah, I naïvely grouped those files together with the documentation when I compiled it myself, due to the similar file endings. Fix coming up.

**Comment by oyvindrobertsen** [comment]
That seems to do the trick on my end.

File: Library/Formula/libghthash.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  sha1 "436146998c30a31b83adf4f58d978fccd78f2ba5"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+    Dir["#{lib}/*[^.a]"].each do |f|
+      File.rename(f, "#{lib}/" + File.basename(f) + ".dylib")
```
Comment (mikemcquaid) [inline Library/Formula/libghthash.rb]: Can you please submit a patch doing to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Comment by vszakats** [comment]
Hi all, if you're at this please update `url` to:
https://www.bth.se/people/ska/sim_home/filer/libghthash-0.6.2.tar.gz

and `homepage` to:
https://www.bth.se/people/ska/sim_home/libghthash.html
(redirects to http for now, but anyway)

**Comment by jacknagel** [comment]
ping

**Pull request closed without merging**